src/content/extension.js

```diff
@@ -1,24 +1,261 @@
-import GitHub from "./github";
-import LeetCode from "./leetCode";
-import StackOverflow from "./stackOverflow";
+/* globals chrome */
 
+import prettier from "prettier/standalone";
 import Storage from "./storage";
+import { PARSERS, PARSERS_LANG_MAP } from "./parsers";
 
-const extensions = [GitHub, StackOverflow, LeetCode];
+const STACKEXCHANGE_SITES = [
+  "https://stackoverflow.com",
+  "https://askubuntu.com",
+  "https://mathoverflow.com",
+  "https://serverfault.com",
+  "https://stackapps.com",
+  "https://superuser.com",
+];
+const STACKEXCHANGE_URL_REGEX = /^https:\/\/([a-z]+).stackexchange.com/;
+const STACKEXCHANGE_VALID_PATHNAMES = /(^\/questions|\/posts\/\d+\/edit|^\/review)/u;
 
-export default class {
+export default class Extension {
   constructor() {
     this._storage = new Storage();
   }
 
   async init() {
     await this._storage.init();
+    this._bindListeners();
+  }
+
+  _bindListeners() {
+    chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
+      if (message.action !== "runPrettierFormat") {
+        return;
+      }
+
+      const inputEl = document.activeElement;
+
+      if (!inputEl) {
+        return;
+      }
+
+      this._format(inputEl);
+      inputEl.focus();
+    });
+  }
+
+  _isStackOverflow() {
+    const origin = window.location.origin;
+
+    return (
+      (STACKEXCHANGE_SITES.some((url) => url === origin) ||
+        STACKEXCHANGE_URL_REGEX.test(origin)) &&
+      STACKEXCHANGE_VALID_PATHNAMES.test(window.location.pathname)
+    );
+  }
+
+  _format(inputEl) {
+    if (this._isStackOverflow()) {
+      this._formatStackOverflow(inputEl);
+    }
+
+    this._formatDefault(inputEl);
+  }
```

Comment:
```suggestion
  _format(inputEl) {
   const options = this._storage.get().prettierOptions;
  
    if (this._isStackOverflow()) {
      this._formatStackOverflow(inputEl, options);
    }

    this._formatDefault(inputEl, options);
  }
```

instead of getting it multiple times during execution (in _formatStackOverflow)

Reply:
Good idea +1: